Jenkinsfile

```diff
@@ -8,18 +8,38 @@ pipeline {
         durabilityHint 'PERFORMANCE_OPTIMIZED'
         timeout(360)
     }
+    parameters {
+        booleanParam defaultValue: false, description: 'Archive 3rd Party Images', name: 'ARCHIVE'
+    }
     triggers {
         issueCommentTrigger('.*^recheck$.*')
     }
     stages {
         stage('Smoke Tests') {
             when {
-                expression { !edgex.isReleaseStream() }
+                expression { !edgex.isReleaseStream() && env.ARCHIVE == 'false' }
             }
             steps {
                 build job: '/edgexfoundry/edgex-taf-pipelines/smoke-test', parameters: [string(name: 'SHA1', value: env.GIT_COMMIT), string(name: 'TEST_ARCH', value: 'All'), string(name: 'WITH_SECURITY', value: 'All')]
             }
         }
+
+        stage('Archive 3rd Party Images') {
+            when {
+                expression { env.ARCHIVE == 'true' }
+            }
+            steps {
+                edgeXDockerLogin(settingsFile: 'ci-build-images-settings')
+                bootstrapBuildX()
+
+                script {
+                    def images = sh(script: "grep image docker-compose.yml | grep -v nexus | awk '{print \$2}'", returnStdout: true).trim()
```

Comment:
Should this be expected to run on the released compose file? If so it should ignore `edgexfoundry` rather than `nexus`

Reply:
It technically can be run prior to release on the main branch. I will update the grep though.